.github/workflows/reusable-solidity-docs.yml

```diff
@@ -0,0 +1,248 @@
+          Usually there shouldn't be need to set this option to `true` if
+          Docgen's `pages` attribute in `hardhat.config.ts` is set to `single`.
+          In other cases the option should be use with caution."
+        type: boolean
+        required: false
+        default: false
+      commentPR:
+        description: "True if you want to add a comment with the path to the
+          generated files in the PR invoking the workflow. If the workflow is
+          not triggered by the `pull_request` event, having this input set to
+          `true` will not brake the execution."
+        type: boolean
+        required: false
+        default: false
+      exportAsGHArtifacts:
+        description: "True if you want to see the artifacts generated on various
+          stages of workflow execution in the GH run details."
+        type: boolean
+        required: false
+        default: false
+    secrets:
+      githubToken:
+        description: "A GitHub API token for the destination organization.
+          Should have `repo` scope. Required if `publish=true`."
+        required: false
+      gpgPrivateKey:
+        description: "A GPG private key needed when `verifyCommits` is set to
+          true. See
+          https://github.com/crazy-max/ghaction-import-gpg#prerequisites for
+          instructions on key generation."
+        required: false
+      gpgPassphrase:
+        description: "A passphrase of the GPG key. Needed when `verifyCommits`
+          is set to true."
+        required: false
+
+jobs:
+  docs-generate-html-and-publish:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: .${{ inputs.projectSubfolder }}
+    steps:
+      - uses: actions/checkout@v3
+
+      # In this step we modify the format of the comments in the Solidity
+      # contracts files. We do that because our original formatting is not
+      # processed by Docgen in the way we would like.
+      # To nicely display lists (like the list of requirements) we need to
+      # remove multiple space chars after the `///` comment. We do that by
+      # executing `sed 's_///[[:blank:]]*_///_'` on all contracts files, which
+      # substitutes `///` + 0 or more spaces/tabs with just `///`.
+
+      - name: Prepare contract files for further processing
+        if: inputs.removeTrailingSpacesInComments
+        shell: bash
+        run: |
+          find ./contracts \
+            -name "*.sol" \
+            -type f \
+            -exec sed -i 's_///[[:blank:]]*_///_' {} \;
+
+      - name: Execute additional command
+        if: inputs.preProcessingCommand != null
+        shell: bash
+        run: ${{ inputs.preProcessingCommand }}
+
+      - name: Export artifacts
+        if: inputs.exportAsGHArtifacts == true
+        uses: actions/upload-artifact@master
+        with:
+          name: contracts-after-preprocessing
+          path: .${{ inputs.projectSubfolder }}/contracts
```

Comment:
Why would we need this? Debugging?

Reply:
Yes, exactly that, it can be helpful with debugging, especially when the action gets introduced in a new project. I was wondering whether to introduce a `debug` input and make the action use `if: inputs.debug == true` instead of `if: inputs.exportAsGHArtifacts == true`. But I decided to limit the number of inputs, as there's already lot of them.
I don't have a strong opinion about this, so if you believe using `debug` input would be better, I can do that.